pkg/health/server/prober.go

```diff
@@ -229,17 +273,24 @@ func (p *prober) setNodes(nodes nodeMap) {
 	}
 }
 
-func (p *prober) httpProbe(node string, ip string, port int) *models.ConnectivityStatus {
+func (p *prober) httpProbe(node *ciliumModels.NodeElement, addr *net.IPAddr, port int) *models.ConnectivityStatus {
 	result := &models.ConnectivityStatus{}
 
-	host := fmt.Sprintf("http://%s:%d", ip, port)
+	host := fmt.Sprintf("http://%s:%d", addr, port)
```

Comment:
TBH, I think that using `JoinHostPort` will be completely counter-productive here. Why? Because that function takes strings as arguments. We have instances of `net.IPAddr` and `int`. Also, we still need to append `http://` prefix. So usage of `JoinHostPort` in this line would look like:

```
host := fmt.Sprintf("http://%s", net.JoinHostPort(addr.String(), strconv.FormatInt(port, 10)))
```

I'm sorry, but IMO this looks completely ugly.